lib/node_modules/@stdlib/blas/base/sdot/lib/sdot.js

```diff
@@ -0,0 +1,97 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2019 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );
+
+
+// VARIABLES //
+
+var M = 5;
+
+
+// MAIN //
+
+/**
+* Computes the dot product of `x` and `y`.
+*
+* @param {PositiveInteger} N Number of values to use from each array
+* @param {Float32Array} x First array
+* @param {PositiveInteger} strideX Stride length on the first array
+* @param {Float32Array} y Second array
+* @param {PositiveInteger} strideY Stride length of the second array
+* @returns {number} The dot product of x and y
+*
+* @example
+* var x = new Float32Array( [ 4.0, 2.0, -3.0, 5.0, -1.0 ] );
+* var y = new Float32Array( [ 2.0, 6.0, -1.0, -4.0, 8.0 ] );
+
+* sdot( x.length, x, 1, y, 1 );
+* // -5.0
+*/
+function sdot( N, x, strideX, y, strideY ) {
+	var dot = 0.0;
+	var ix;
+	var iy;
+	var m;
+	var i;
+
+	if ( N <= 0 ) {
+		return dot;
+	}
+	if ( strideX === 1 && strideY === 1 ) {
+		m = N % M;
+		for ( i = 0; i < m; i++ ) {
+			dot += float64ToFloat32( x[ i ] * y[ i ] );
+		}
+		if (N < M) {
+			return dot;
+		}
+		for (i = m; i < N; i += M) {
+			dot = float64ToFloat32( dot +
+				( x[ i ] * y[ i ] ) +
+				( x[ i+1 ] * y[ i+1 ] ) +
+				( x[ i+2 ] * y[ i+2 ] ) +
+				( x[ i+3 ] * y[ i+3 ] ) +
+				( x[ i+4 ] * y[ i+4 ] ) );
+		}
+		return dot;
+	}
+	if ( strideX < 0 ) {
+		ix = ( 1-N ) * strideX;
+	} else {
+		ix = 0;
+	}
+	if ( strideY < 0 ) {
+		iy = ( 1-N ) * strideY;
+	} else {
+		iy = 0;
+	}
+	for ( i = 0; i < N; i++ ) {
+		dot += float64ToFloat32( x[ ix ] * y[ iy ] );
```

Comment:
```suggestion
		dot = float64ToFloat32( dot + (x[ ix ] * y[ iy ]) );
```

For the same reasons as the other file.